pkg/network/ebpf/c/prebuilt/usm.c

```diff
@@ -54,25 +54,40 @@ SEC("uprobe/SSL_do_handshake")
 int uprobe__SSL_do_handshake(struct pt_regs* ctx) {
     u64 pid_tgid = bpf_get_current_pid_tgid();
     void *ssl_ctx = (void *)PT_REGS_PARM1(ctx);
+    ssl_handshake_args_t args = { 0 };
+    args.ctx = ssl_ctx;
     log_debug("uprobe/SSL_do_handshake: pid_tgid=%llx ssl_ctx=%llx\n", pid_tgid, ssl_ctx);
-    bpf_map_update_with_telemetry(ssl_ctx_by_pid_tgid, &pid_tgid, &ssl_ctx, BPF_ANY);
+    bpf_map_update_with_telemetry(ssl_ctx_by_pid_tgid, &pid_tgid, &args, BPF_ANY);
     return 0;
 }
 
 SEC("uretprobe/SSL_do_handshake")
 int uretprobe__SSL_do_handshake(struct pt_regs* ctx) {
     u64 pid_tgid = bpf_get_current_pid_tgid();
+    int ret = (int)PT_REGS_RC(ctx);
     log_debug("uretprobe/SSL_do_handshake: pid_tgid=%llx\n", pid_tgid);
-    bpf_map_delete_elem(&ssl_ctx_by_pid_tgid, &pid_tgid);
+    if (ret >= 0) {
+        bpf_map_delete_elem(&ssl_ctx_by_pid_tgid, &pid_tgid);
+        return 0;
+    }
+
+    // asynchronous mode probably
+    ssl_handshake_args_t *args = bpf_map_lookup_elem(&ssl_ctx_by_pid_tgid, &pid_tgid);
+    if (args == NULL) {
+        return 0;
+    }
+    args->timestamp = bpf_ktime_get_ns();
     return 0;
```

Comment:
where do we clean the `ssl_ctx_by_pid_tgid` map in an async mode?

Reply:
[here](https://github.com/DataDog/datadog-agent/pull/16439/files#diff-cba5ebc99dec7b2e068282aaf0b1a6aec58713e71bf2da9aacd89a3ef3f87a33R161)
in map_ssl_ctx_to_sock() called by tcp_sendmsg